internal/rollout/device_selection/reconciler.go

```diff
@@ -134,33 +153,37 @@ func (r *reconciler) reconcileFleet(ctx context.Context, orgId uuid.UUID, fleet
 
 		// Once the batch is complete, set the success percentage of the current batch
 		if err = selection.SetCompletionReport(ctx); err != nil {
-			r.log.WithError(err).Errorf("%v/%s: SetCompletionReport", orgId, lo.FromPtr(fleet.Metadata.Name))
+			r.sendLog(orgId, fleetName, "SetCompletionReport", err)
 			break
 		}
 		hasMoreSelections, err := selector.HasMoreSelections(ctx)
 		if err != nil {
-			r.log.WithError(err).Errorf("%v/%s: HasMoreSelections", orgId, lo.FromPtr(fleet.Metadata.Name))
+			r.sendLog(orgId, fleetName, "HasMoreSelections", err)
 			break
 		}
 		if !hasMoreSelections {
 			if err = selection.OnFinish(ctx); err != nil {
```

Comment:
Maybe emit event when there are no more selections? It means fleet rollout was complete